pkg/binding/controller.go

```diff
@@ -124,6 +136,50 @@ func NewController(parentLogger logr.Logger, wdsRestConfig *rest.Config, imbsRes
 	return makeController(parentLogger, dynamicClient, kubernetesClient, extClient, ocmClientset, ocmClient, wdsName, allowedGroupsSet)
 }
 
+func countGVRsViaAggregatedDiscovery(countClient kubernetes.Interface) (int, error) {
+	dc := countClient.Discovery().(*discovery.DiscoveryClient)
+	if dc.UseLegacyDiscovery { // by default it should be false already, just double check
+		dc.UseLegacyDiscovery = false
+	}
+	apiResourceLists, err := dc.ServerPreferredResources()
```

Comment:
This can be expensive! Wouldn't it be better to find a way to do it only once?

Reply:
Not sure I understand the question. This PR is doing exactly reducing the number of calls to `ServerPreferredResources()` (during CRD handling) to only one. And the only one is right here.

Reply:
I see that func also being called at https://github.com/kubestellar/kubestellar/blob/48f75f51b37bd325ad4c075aa15ad0c6b2158d6b/pkg/binding/controller.go#L249 and https://github.com/kubestellar/kubestellar/blob/48f75f51b37bd325ad4c075aa15ad0c6b2158d6b/pkg/crd/crds.go#L94